src/clients/base-server-client.js

```diff
@@ -0,0 +1,160 @@
+    }
+  }
+
+  /**
+   * @param  {object} [parameters] The query parameters as an object (optional).
+   *
+   * @return {promise} The request promise.
+   */
+  search (parameters = {}) {
+    const url = this.__buildUrl(null, parameters)
+    return this.sendRequest(url)
+  }
+
+  /**
+   * @param  {integer} [id] The resource id.
+   * @param  {object} [parameters] The query parameters as an object (optional).
+   *
+   * @return {promise} The request promise.
+   */
+  find (id, parameters = {}) {
+    const url = this.__buildUrl(id, parameters)
+    return this.sendRequest(url)
+  }
+
+  /**
+   * @param  {object} [data] The data object / attribute hash.
+   * @param  {object} [parameters] The query parameters as an object (optional).
+   *
+   * @return {promise} The request promise.
+   */
+  create (data, parameters = {}) {
+    const url = this.__buildUrl(null, parameters)
+
+    return this.sendRequest(url, {
+      body: data,
+      method: 'POST'
+    })
+  }
+
+  /**
+   * @param  {integer} [id] The resource id.
+   * @param  {object} [data] The data object / attribute hash.
+   * @param  {object} [parameters] The query parameters as an object (optional).
+   *
+   * @return {promise} The request promise.
+   */
+  update (id, data, parameters = {}) {
+    const url = this.__buildUrl(id, parameters)
+
+    return this.sendRequest(url, {
+      body: data,
+      method: 'PUT'
+    })
+  }
+
+  /**
+   * @param  {integer} [id] The resource id.
+   * @param  {object} [parameters] The query parameters as an object (optional).
+   *
+   * @return {promise} The request promise.
+   */
+  destroy (id, parameters = {}) {
+    const url = this.__buildUrl(id, parameters)
+    return this.sendRequest(url, {
+      method: 'DELETE'
+    })
+  }
+
+  /**
+   * Makes an http request and returns a promise.
+   *
+   * @param  {string} [url] the url
+   * @param  {object} [options] the options we want to pass to 'fetch'
+   *   - body   can be null
+   *   - method must be GET, POST, PUT, PATCH or DELETE. Default is GET.
+   *
+   * @return {promise} the sendRequest promise
+   */
+  sendRequest (url, options = {}) {
+    const opts = {
+      headers: {
+        'Content-Type': this.config.contentType
+      },
+      method: (options.method || 'GET').toUpperCase()
+    }
+
+    if (options.body) {
+      opts.body = JSON.stringify(options.body)
+    }
+
+    if (this.config.authorizationRequired) {
+      const tokenClient = new ServiceTokenClient({
+        accessKey: this.config.accessKey,
+        secretKey: this.config.secretKey,
+        baseUrl: this.config.doorkeeperBaseUrl
+      })
+
+      return tokenClient.createServiceToken({}).then(token => {
+        opts.headers['Authorization'] = `Bearer ${token.jwt}`
+        return simpleFetch(url, opts)
+      })
+    } else {
+      opts.credentials = 'same-origin'
+      return simpleFetch(url, opts)
+    }
+  }
+
+  __buildUrl (path, parameters) {
```

Comment:
since this and other some methods are:
https://github.com/infopark-customers/bima-shark-js-sdk/blob/493e34acf66b3805dc979dee2478011e0396bfe0/src/clients/base-client.js#L177
would be great to extract some base class for both of these and then `extend`, so we skip copy/pasta